### PR TITLE
Add S3 Bucket Rules W/Master Key Null query #1246

### LINKED ISSUE #1246: Add S3 Bucket Rules With Master Key Id Null query for Ansible
### Platform
Ansible

### Description
If algorithm is AES256 then the master key is null, empty or undefined, otherwise the master key is required

### PR DESCRIPTION
**Pull request by joelcarvalhocheckmarx**
If algorithm is AES256 then the master key is null, empty or undefined, otherwise the master key is required

Closes #1246

### CODE REVIEW

File: assets/queries/ansible/s3_bucket_rules_with_awskms/query.rego [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+CxPolicy [ result ] {
+   document := input.document[i]
+   tasks := getTasks(document)
+   task := tasks[t]
+   s3_bucket := task["amazon.aws.s3_bucket"]
+   encryption := s3_bucket.encryption
+   
+   not encryption == "AES256"
+   check_key_id(s3_bucket)
```
Comment (joaomartinscx) [inline assets/queries/ansible/s3_bucket_rules_with_awskms/query.rego:11]: The query description leads me to think that when the encryption algorithm is AES256, the master key should be empty/undefined, but you are not verifying it. Should you be?

**Review by joaomartinscx — CHANGES_REQUESTED** [review]

**Comment by markmishaevcx** [comment]
Scan submitted to Checkmarx

**Comment by markmishaevcx** [comment]
Logo<br />**Checkmarx SAST - Scan Summary & Details**
### Cx-SAST Summary
Total of 4 vulnerabilities<br />High&nbsp;**0 High**<br />Medium&nbsp;**3 Medium**<br />Low&nbsp;**1 Low**<br />Info&nbsp;**0 Info**<br />
### Violation Summary
Medium&nbsp;**3** **Medium**<br />
[View more details on Checkmarx UI](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049087&projectid=735)
### Cx-SAST Details
|Lines|Severity|Category|File|Link|
---|---|---|---|---
[30](https://github.com/Checkmarx/kics/blob/feature/add_query_ansible_s3_bucket_rules_with_awskms/cmd/builder/main.go#L30) |Medium|Path_Traversal|cmd/builder/main.go|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049087&projectid=735&pathid=3)
[80](https://github.com/Checkmarx/kics/blob/feature/add_query_ansible_s3_bucket_rules_with_awskms/test/main_test.go#L80) |Medium|Denial_Of_Service_Resource_Exhaustion|test/main_test.go|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049087&projectid=735&pathid=2)
[52](https://github.com/Checkmarx/kics/blob/feature/add_query_ansible_s3_bucket_rules_with_awskms/pkg/kics/service.go#L52) |Medium|Denial_Of_Service_Resource_Exhaustion|pkg/kics/service.go|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049087&projectid=735&pathid=1)

**Review by joaomartinscx — APPROVED** [review]
Everything looks okay, good job!

**Pull request merged**